encfs/RawFileIO.cpp

```diff
@@ -240,13 +244,16 @@ bool RawFileIO::write(const IORequest &req) {
     RLOG(ERROR) << "Write error: wrote " << req.dataLen - bytes << " bytes of "
                 << req.dataLen << ", max retries reached";
     knownSize = false;
-    return false;
-  }
-  if (knownSize) {
-    off_t last = req.offset + req.dataLen;
-    if (last > fileSize) {
-      fileSize = last;
+    return (eno) ? -eno : -EIO;
+  } else {
```

Comment:
clang-tidy will complain about this.  No need to have an `else` clause after a block that returns.

Please remove the else bracket,
 ie `} else { if (knownSize) ...` -> `} \n if (knownSize) ...`